qiskit/pulse/instruction_schedule_map.py

```diff
@@ -161,39 +169,89 @@ def get(self,
 
         Returns:
             The Schedule defined for the input.
+
+        Raises:
+            PulseError: When invalid parameters are specified.
         """
         instruction = _get_instruction_string(instruction)
         self.assert_has(instruction, qubits)
-        schedule_generator = self._map[instruction].get(_to_tuple(qubits))
+        schedule_param_tuple = deepcopy(self._map[instruction][_to_tuple(qubits)])
+
+        # create parameter-value mapping
+        if len(params) > len(schedule_param_tuple.parameters):
+            raise PulseError('Too many values to bind: {}.'.format(', '.join(map(str, params))))
+
+        if not all(key in schedule_param_tuple.parameters for key in kwparams.keys()):
+            raise PulseError('Keys not defined: {}'.format(', '.join(kwparams.keys())))
+
+        bind_parameters = dict(zip_longest(schedule_param_tuple.parameters, params))
+        bind_parameters.update(kwparams)
 
-        if callable(schedule_generator):
-            return schedule_generator(*params, **kwparams)
-        # otherwise this is just a Schedule
-        return schedule_generator
+        # callback function
+        if callable(schedule_param_tuple.schedule):
+            return schedule_param_tuple.schedule(**bind_parameters)
+
+        # schedule
+        sched_params = list(schedule_param_tuple.schedule.parameters)
+        if sched_params:
```

Comment:
```suggestion
        sched = schedule_param_tuple.schedule
        if sched.is_parameterized():
```

Reply:
Recommending this pair of changes to make it more clear that we are not looking at `schedule_param_tuple.parameters`